src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,310 @@
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.EVAL_DATA_DIR
+        logger.debug(
+            "Output dir provided is not a directory or does not exist, writing results to default evaluation output directory %s",
+            output_dir,
+        )
+
+    return f"{output_dir}/responses-{model_name}-{timestamp}.{file_type}"
+
+
+def get_api_base_for_model(
+    ctx: click.Context,
+    model: str,
+    temperature: float,
+    model_prompt: str,
+    max_workers: str | int | None,
+    gpus: int | None,
+    backend: str | None,
+    enable_serving_output: bool,
+) -> tuple:
+    server = None
+    if model_is_localand_valid(model):
+        logger.debug("DK-Bench model is local")
+        model_name = get_model_name(model)
+        server, api_base, _ = launch_server(
+            eval_serve=ctx.obj.config.serve,
+            tls_client_cert=ctx.params["tls_client_cert"],
+            tls_client_key=ctx.params["tls_client_key"],
+            tls_client_passwd=ctx.params["tls_client_passwd"],
+            tls_insecure=ctx.params["tls_insecure"],
+            model=model,
+            model_name=model_name,
+            max_workers=max_workers,
+            gpus=gpus,
+            backend=backend,
+            enable_serving_output=enable_serving_output,
+        )
+    else:
+        logger.debug("DK-Bench model is an endpoint")
+        http_params = http_client(
+            {
+                "tls_client_cert": ctx.params["tls_client_cert"],
+                "tls_client_key": ctx.params["tls_client_key"],
+                "tls_client_passwd": ctx.params["tls_client_passwd"],
+                "tls_insecure": ctx.params["tls_insecure"],
+            }
+        )
+
+        api_key = os.environ.get("STUDENT_ENDPOINT_API_KEY", None)
+        if api_key is None:
+            logger.debug(
+                "API_KEY for model at endpoint %s is not set. To set it set the environment variable $STUDENT_ENDPOINT_API_KEY",
+                model,
+            )
+
+        model_name = get_endpoint_model_name(model, api_key, http_params)
+        api_base = model
+
+    openai_client = get_openai_client(model_api_base=api_base, api_key=api_key)
+    model_config = ModelConfig(
+        model_name=model_name, temperature=temperature, system_prompt=model_prompt
+    )
+
+    return server, model_config, openai_client
+
+
+def make_run_dir(output_dir: str) -> str:
+    now = datetime.now()
+    timestamp = now.strftime("%m-%d-%Y_%H-%M-%S-%f")
+    if output_dir.endswith("/"):
+        output_dir = output_dir[:-1]
+    run_dir = f"{output_dir}/job-{timestamp}"
+
+    path_does_not_exist = not os.path.exists(output_dir)
+    path_is_not_dir = not os.path.isdir(output_dir)
+    if path_does_not_exist or path_is_not_dir:
+        run_dir = f"{DEFAULTS.EVAL_DATA_DIR}/job-{timestamp}"
+
+    os.makedirs(run_dir, exist_ok=True)
+    return run_dir
+
+
+def print_header():
```

Comment:
Were you planning on expanding this function?

Reply:
This is just the beginning of the output. I'll get rid of the function for the time being since it's so small.